druid/examples/transparency.rs

```diff
@@ -0,0 +1,89 @@
+// Copyright 2019 The Druid Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! An example of a transparent window background.
+//! Useful for dropdowns, tooltips and other overlay windows.
+
+use druid::{kurbo::{Circle}, widget::{Button, Flex}};
+use druid::widget::prelude::*;
+use druid::{
+    AppLauncher, Color, LocalizedString, Rect,
+    WindowDesc,
+};
+
+struct CustomWidget;
+
+impl Widget<String> for CustomWidget {
+    fn event(&mut self, _ctx: &mut EventCtx, _event: &Event, _data: &mut String, _env: &Env) {}
+
+    fn lifecycle(
+        &mut self,
+        _ctx: &mut LifeCycleCtx,
+        _event: &LifeCycle,
+        _data: &String,
+        _env: &Env,
+    ) {
+    }
+
+    fn update(&mut self, _ctx: &mut UpdateCtx, _old_data: &String, _data: &String, _env: &Env) {}
+
+    fn layout(
+        &mut self,
+        _layout_ctx: &mut LayoutCtx,
+        bc: &BoxConstraints,
+        _data: &String,
+        _env: &Env,
+    ) -> Size {
+
+        if bc.is_width_bounded() | bc.is_height_bounded() {
+            let size = Size::new(100.0, 100.0);
+            bc.constrain(size)
+        } else {
+            bc.max()
+        }
+    }
+
+    // The paint method gets called last, after an event flow.
+    // It goes event -> update -> layout -> paint, and each method can influence the next.
+    // Basically, anything that changes the appearance of a widget causes a paint.
+    fn paint(&mut self, ctx: &mut PaintCtx, data: &String, env: &Env) {
```

Comment:
put underscores before unused parameters

Reply:
This was patched by Robert.